src/components/Form Fields/dropdown-headless.tsx

```diff
@@ -0,0 +1,116 @@
+import { Fragment } from "react";
+import { Menu, Transition } from "@headlessui/react";
+import {
+  FieldValues,
+  useController,
+  UseControllerProps
+} from "react-hook-form";
+import { HiCheck, HiChevronDown } from "react-icons/hi";
+
+import Label from "@/components/Form Fields/box-label";
+import Error from "@/components/Form Fields/error-msg";
+
+export interface FormProps<T extends FieldValues = FieldValues>
+  extends UseControllerProps<T> {
+  options: { id: number; text: string }[];
+  width?: number;
+  height?: number;
+  required?: boolean;
+  placeholder?: string;
+  label?: string;
+}
+
+function classNames(...classes: any) {
+  return classes.filter(Boolean).join(" ");
+}
+
+/*
+This is a component for a dropdown menu
+Will display a dropdown displaying options inputted
+Input:
+  options: { id: number; text: string }[]; // array of objects with option text and id number
+  placeholder: string; // placeholder string before any option is selected
+  label?: string; // text on border of button
+Output:
+  A dropdown that is compatible w/ React-hook-forms 
+*/
+
+export default function DropDown<T extends FieldValues = FieldValues>({
+  options,
+  placeholder = "-select-",
+  label = "",
+  ...props
+}: FormProps<T>) {
+  const { field, fieldState } = useController(props);
+
+  const errorStyle = !fieldState.invalid
```

Comment:
A lot of this is the same. Only difference I see is the ring colour and size (`ring-grey-300` for valid states and `ring-red-500` for invalid states. We can store most of it in a constant variable and add the above when necessary.

Also, a few readability things to consider:
- Why do we have the ternary condition set to `!fieldState.invalid`? It reads "If not invalid, then have valid style, otherwise have invalid style"
    - A better way to approach this is to get rid of the double negatives and simplify as much as possible. See example below
- Another concern is the name of the variable name is `errorStyle` but we won't always have an 'error style'. To add to that, the Error component will only show if `field.invalid` is true, so 'errorStyle' is not great because it implies we are in an error state 100% of the time

I suggest define a few variables to help organise this a bit better.

First I would have the two differentiating style properties stored as their own constants
```ts
const baseStyle = `flex h-10 w-full justify-between overflow-hidden rounded-lg bg-white px-3 py-2.5 text-sm font-medium text-gray-900 shadow-sm ring-1 ring-inset hover:shadow-grey-300 `;

const validStateStyle = `ring-grey-300`
const errorStyle = `ring-red-500`;
```
This also helps if you ever have to change the base styling of the button. You will only need to change something in one place rather than two.

Next, I would change the ternary operator to read "If invalid, use invalid state, else use normal state". Simpler and easier to read.
```ts
let menuButtonStyle = baseStyle;
menuButtonStyle += field.invalid
    ? errorStyle
    : validStateStyle;
```

Let me know what you think. (excuse my Typescript if it is not syntactically correct)

Reply:
Yeah, it is a bit messy. Thanks for the suggestions!

I've refactored the styling for changing the border colors into variables.

I've also changed the logic for the ternary operators.

Hopefully it's a bit more readable now.